wire/msgtx.go

```diff
@@ -391,24 +392,28 @@ func (msg *MsgTx) serialize(serType TxSerializeType) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-// mustSerialize returns the serialization of the transaction for the provided
-// serialization type without modifying the original transaction.  It will panic
-// if any errors occur.
-func (msg *MsgTx) mustSerialize(serType TxSerializeType) []byte {
-	serialized, err := msg.serialize(serType)
+// mustHash returns the hash of the transaction for the provided
+// serialization type without modifying the original transaction.
+// It will panic if serialization fails.
+func (msg *MsgTx) mustHash(hasher *blake256.Hasher256, serType TxSerializeType) chainhash.Hash {
+	// Shallow copy so the serialization type can be changed without
+	// modifying the original transaction.
+	mtxCopy := *msg
+	mtxCopy.SerType = serType
+	err := mtxCopy.Serialize(hasher)
 	if err != nil {
 		panic(fmt.Sprintf("MsgTx failed serializing for type %v",
 			serType))
 	}
-	return serialized
+	return hasher.Sum256()
 }
 
 // TxHash generates the hash for the transaction prefix.  Since it does not
 // contain any witness data, it is not malleable and therefore is stable for
 // use in unconfirmed transaction chains.
 func (msg *MsgTx) TxHash() chainhash.Hash {
 	// TxHash should always calculate a non-witnessed hash.
-	return chainhash.HashH(msg.mustSerialize(TxSerializeNoWitness))
+	return msg.mustHash(blake256.NewHasher256(), TxSerializeNoWitness)
```

Comment:
The same applies regarding commenting the func that the hash function now is explicitly BLAKE-256 since it no longer follows what `chainhash` does.